src/Stratis.Bitcoin.IntegrationTests/BlockStore/ReOrgRegularlySpecification_Steps.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System.Linq;
+using FluentAssertions;
+using FluentAssertions.Common;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Consensus;
+using Stratis.Bitcoin.Features.Wallet;
+using Stratis.Bitcoin.Features.Wallet.Controllers;
+using Stratis.Bitcoin.Features.Wallet.Models;
+using Stratis.Bitcoin.IntegrationTests.EnvironmentMockUpHelpers;
+using Xunit.Abstractions;
+
+namespace Stratis.Bitcoin.IntegrationTests.BlockStore
+{
+    public partial class ReOrgRegularlySpecification
+    {
+        private NodeBuilder nodeBuilder;
+        private CoreNode selfishMiner;
+        private CoreNode secondNode;
+        private CoreNode thirdNode;
+        private CoreNode fourthNode;
+        private SharedSteps sharedSteps;
+        private Transaction secondNodeTransaction;
+        private Money secondNodeTransationFee;
+        private int selfishBlockHeight;
+        private const string AccountZero = "account 0";
+        private const string WalletZero = "wallet 0";
+        private const string WalletPassword = "123456";
+
+        public ReOrgRegularlySpecification(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        protected override void BeforeTest()
+        {
+            this.sharedSteps = new SharedSteps();
+        }
+
+        protected override void AfterTest()
+        {
+        }
+
+        private void four_nodes()
+        {
+            this.nodeBuilder = NodeBuilder.Create();
+
+            this.selfishMiner = this.nodeBuilder.CreateStratisPowNode();
+            this.secondNode = this.nodeBuilder.CreateStratisPowNode();
+            this.thirdNode = this.nodeBuilder.CreateStratisPowNode();
+            this.fourthNode = this.nodeBuilder.CreateStratisPowNode();
+
+            this.nodeBuilder.StartAll();
+
+            this.selfishMiner.NotInIBD();
+            this.secondNode.NotInIBD();
+            this.thirdNode.NotInIBD();
+            this.fourthNode.NotInIBD();
+
+            this.selfishMiner.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+            this.secondNode.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+            this.thirdNode.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+            this.fourthNode.FullNode.WalletManager().CreateWallet(WalletPassword, WalletZero);
+
+            this.selfishMiner.CreateRPCClient().AddNode(this.secondNode.Endpoint, true);
```

Comment:
It will be faster if you call addnode for all the nodes first and then wait on each node.